api/grpcserver/grpcserver_test.go

```diff
@@ -466,7 +466,28 @@ func launchServer(t *testing.T, services ...ServiceAPI) func() {
 		cfg.StartNodeService,
 		cfg.StartSmesherService,
 		cfg.StartTransactionService)
-	time.Sleep(3 * time.Second) // wait for server to be ready (critical on CI)
+
+	timer := time.NewTimer(3 * time.Second)
+	defer timer.Stop()
+
+	// wait for server to be ready (critical on CI)
+	// 3 possible cases:
+	// - timed out
+	// - grpcStarted, then jsonStarted or timed out
+	// - jsonStarted, then grpcStarted or timed out
+	select {
+	case <-timer.C:
+	case <-grpcStarted:
+		select {
+		case <-timer.C:
+		case <-jsonStarted:
+		}
+	case <-jsonStarted:
+		select {
+		case <-timer.C:
+		case <-grpcStarted:
+		}
+	}
```

Comment:
it doesn't matter which one we will select first? in that case you could iterate over array with channels. it will make a bit prettier i believe
```go
for _, start := range []chan struct{}{json, grpc} {
  select {
  ...
  }
}
```

Reply:
@dshulyak I agree that this doesn't look pretty. 

No, the order does not matter. What we need is to wait until both channels are closed or until a 3-second timeout. I'm not sure I'm following your thought about `range`. Could you please add the implementation of the `select` clause body?

Reply:
nothing changes in the select
```go

	for _, ch := range []chan struct{}{grpc, json} {
		select {
		case <-ch:
		case <-timer.C:
			panic("timeout")
		}
	}
```

Reply:
Now I see. So many `select`s are not needed indeed. Thanks